corehq/util/couchdb_management.py

```diff
@@ -36,6 +36,12 @@ def all_dbs_by_slug(self):
         return {slug: Database(db_uri)
                 for slug, db_uri in self.all_db_uris_by_slug.items()}
 
+    @property
+    @memoized
+    def all_dbs_by_name(self):
```

Comment:
nitpick, but with the three different representations of a db flying around (slug, db_uri, and db_name), the functions can get a little confusing. I like the name you chose for `get_db_for_db_name`—could you also call this `all_dbs_by_db_name`?